File: Linked  List/203. Remove Linked List Elements/Solution.java
/*
 时间复杂度 O(n)
 空间复杂度 O(1) 

 设立虚拟头节点
*/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummynode = new ListNode(0);
        dummynode.next = head;
        ListNode cru = dummynode;
        
        while(cru.next != null){
            if(cru.next.val == val){
                ListNode delnode = cru.next;
                cru.next = delnode.next;
                delnode.next = null;
                
            }else{
                cru = cru.next;
            }
        }
        ListNode retnode = dummynode.next;
        dummynode.next = null;
        
        return retnode;

    }
}

File: Linked  List/203. Remove Linked List Elements/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head
        cru = dummy
        while cru and cru.next:
        	if cru.next.val == val:
        		cru.next = cru.next.next
        	else:
        		cru = cru.next
        return dummy.next